roles/cluster_api/tasks/patch.yml

```diff
@@ -52,3 +52,17 @@
     - {"crd": "kubeadmconfigtemplates.bootstrap.cluster.x-k8s.io", "path": "/spec/versions/2/schema/openAPIV3Schema/properties/spec/properties/template/properties/spec/properties/joinConfiguration/properties/nodeRegistration/properties/imagePullPolicy/default"}
     - {"crd": "kubeadmcontrolplanes.controlplane.cluster.x-k8s.io", "path": "/spec/versions/2/schema/openAPIV3Schema/properties/spec/properties/kubeadmConfigSpec/properties/initConfiguration/properties/nodeRegistration/properties/imagePullPolicy/default"}
     - {"crd": "kubeadmcontrolplanes.controlplane.cluster.x-k8s.io", "path": "/spec/versions/2/schema/openAPIV3Schema/properties/spec/properties/kubeadmConfigSpec/properties/joinConfiguration/properties/nodeRegistration/properties/imagePullPolicy/default"}
+
+- name: Install required controllers
+  ansible.builtin.command:
+    argv:
+      - kubectl
+      - apply
+      - --server-side
+      - -f
+      - "{{ cluster_api_controller_path }}/{{ cluster_api_infrastructure_provider }}-resource-controller/v{{ cluster_api_openstack_controller_version }}/install.yaml"
+  changed_when: false
+  run_once: true
+  when:
+    - cluster_api_infrastructure_provider == "openstack"
+    - cluster_api_infrastructure_version is ansible.builtin.version('0.12.0', 'ge')
```

Comment:
This workaround here would be to add a `--force` which is okay because the manager of some of those resources is being "taken" over by the ORC instead of CAPO.

However, there are a few things that we need to account for:

- We can simply use `kubernetes.core.k8s` to apply things more natively
- We should move this to the start of the file since we want to adjust the node selector for it.